### PR TITLE
[api][workflow][SIMS #1081] Reassessment workflows

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- Fixed an issue on student appeal where the offering id was not copied from the current assessment.
- Added the _applicationId_ to the assessment data;
- Removed the _applicationId_ from the _assessment-gateway.bpmn_ start variables (only the _assessmentId_ is needed);
- Changed the _assessment-gateway_ to allow the processing of the reassessment;
- The workflow review can also be done on a call to avoid going over every single like changed.

**Main changes on the _assessment-gateway_**

image

**Students Appeals processing**

image

image

### CODE REVIEW

File: sources/packages/workflow/BPMN/assessment-gateway.bpmn [hunk 1]
```diff
@@ -336,188 +242,587 @@ createVariablesFromPayload(output.prop("programYear"));</camunda:script>
...
-      <bpmn:incoming>Flow_1gq6278</bpmn:incoming>
+    <bpmn:intermediateThrowEvent id="Event_0gpagjg" name="Notify error">
+      <bpmn:extensionElements />
+      <bpmn:incoming>Flow_0v56rad</bpmn:incoming>
+      <bpmn:outgoing>Flow_1dd2smr</bpmn:outgoing>
+      <bpmn:messageEventDefinition id="MessageEventDefinition_0os8qcn">
+        <bpmn:extensionElements>
+          <camunda:connector>
+            <camunda:inputOutput>
+              <camunda:inputParameter name="to">gurumoorthy.mohan@gov.bc.ca</camunda:inputParameter>
```
Comment (ann-aot) [inline sources/packages/workflow/BPMN/assessment-gateway.bpmn:314]: not sure, if this is part of your PR. just a thought, we can add this email address as a dynamic variable right

Comment (andrewsignori-aot) [reply sources/packages/workflow/BPMN/assessment-gateway.bpmn:314]: Yes, we can, but not part of the PR. The entire error handling part must be changed.

**Review by ann-aot — APPROVED** [review]
LGTM 👍 Maybe once other devs are back, we can sync up and you can walk us through the workflows

**Comment by andrewsignori-aot** [comment]
> LGTM 👍 Maybe once other devs are back, we can sync up and you can walk us through the workflows

Yes, totally agree. I still need to work on the NOA visualization for the Ministry as the last part of this effort, but a sync-up maybe would be good today.

**Review by guru-aot — APPROVED** [review]
LGTM, Nice and exhaustive work @andrewsignori-aot .

**Review by dheepak-aot — APPROVED** [review]
LGTM @andrewsignori-aot 👍

**Pull request merged**